### ISSUE TITLE
Filter response address records by network interface 

### ISSUE THREAD

**Issue by cyanfish** [jdomnitz/net-mdns#40]
Per https://datatracker.ietf.org/doc/html/rfc6762#section-6.2:
> When a Multicast DNS responder sends a Multicast DNS response message
   containing its own address records, it MUST include all addresses
   that are valid on the interface on which it is sending the message,
   and MUST NOT include addresses that are not valid on that interface
   (such as addresses that may be configured on the host's other
   interfaces).

I have this [mostly working](https://github.com/cyanfish/net-mdns/commit/96327fe217303540d7bcf2eead55d93f53dcf137) in my fork but I'm not sure it's the best way to handle this.

**Comment by jdomnitz** [comment]
I agree this will need some work but it's worthwhile.  I'll work through some other concepts. This is the last area I've found where not complying with the standard causes problems.

**Comment by jdomnitz** [comment]
Reference here: https://github.com/richardschneider/net-mdns/issues/25

Current Issues:
* IPv6 queries should return IPv4 addresses and vice-versa (They may be routable)
* Removing all but the sending NIC breaks support for multi-homing and shared services
* This only works with semetric routing

Easy first steps:
* Loopback addresses are never routable externally
* Update service records in all instances where they may change (WIP)

TBD - How does mDNSResponder handle this?

**Comment by jalfd** [comment]
> Removing all but the sending NIC breaks support for multi-homing and shared services

Are you sure this is a problem? If the service is reachable over multiple different networks, isn't it the responsibility of the client to query on each of its interfaces? So if the client can potentially reach the service on both network A and B it sends separate queries on each network. When this library receives the query on interface A it replies with information specific to that network, and when it receives the query on interface B it replies with information specific to that network. If the service is reachable on both networks, the client will receive both replies. I don't see how else the behavior can be aligned with the RFC quote above, and I would expect it to work for multi-homing too.

> This only works with semetric routing

Again, is that a problem? mdns is only supposed to work with link-local endpoints, isn't it? Is asymmetric routing really going to be a problem in such a scenario?

I'm fairly new to the mdns protocol though, so it is entirely possible that I'm missing something, but I'd be happy to work on a PR to fix this issue
